UI/DamageIndicator.cs

```diff
@@ -0,0 +1,44 @@
+using Godot;
+using System;
+
+public class DamageIndicator : TextureRect
+{
+    // Angle the bullet came from
+    Vector2 ShotFirePosition;
+    float OriginalAngle;
+    public override void _Ready()
+    {
+        // Start transparency Tween in _Ready so we know the Tween has been created
+        Tween TransparencyTween = GetNode<Tween>("Tween");
+        // Tween the "modulate" property
+        TransparencyTween.InterpolateProperty(this,"modulate", new Color(255,255,255,1), new Color(255,255,255,0), 3, Tween.TransitionType.Linear, Tween.EaseType.InOut);
+        TransparencyTween.Start();
+    }
+
+
+    public void SetTargetAngle(Vector2 ShotFirePosition)
+    {
+        // The rotation we're looking at is the Y rotation
```

Comment:
This comment explains something without context, how does the apply to the following lines? It is probably referring to the set to OriginalAngle and so should be next to it. Also we don't look at the Y rotation, the player's look rotates around the Y axis. Once in the correct context the argument could be made that it is uneeded.